src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T item) {
+            this.item = item;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> temporaryItem = tail;
+        Node<T> newNode = new Node<>(value);
+        tail = newNode;
+        if (temporaryItem == null) {
+            head = newNode;
+        } else {
+            temporaryItem.next = newNode;
+            newNode.prev = temporaryItem;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException();
```

Comment:
The check for the index can be extracted into a separate method to avoid duplication (e.g., `checkIndex`).